sway/tree/container.c

```diff
@@ -378,3 +380,26 @@ void container_map(swayc_t *container, void (*f)(swayc_t *view, void *data), voi
 		f(container, data);
 	}
 }
+
+/**
+ * Get a list of containers that are descendents of the container in rendering
+ * order
+ */
+list_t *container_list_children(swayc_t *con) {
+	list_t *list = create_list();
+	list_t *queue = create_list();
+
+	list_add(queue, con);
+
+	swayc_t *current = NULL;
+	while (queue->length) {
+		current = queue->items[0];
+		list_del(queue, 0);
+		list_add(list, current);
+		// TODO floating containers
+		list_cat(queue, current->children);
+	}
+
+	list_free(queue);
+	return list;
+}
```

Comment:
Uh, why is this allocating memory? Maybe make this generator-like and instead of returning a list, take a callback?

Reply:
What about making it one of those cool things with a block like `wl_list_for_each()`?

Reply:
While I think that heavy reliance on wl_list is appropriate for wlroots, it's not how Sway was written and I don't want to introduce new patterns. The map approach is more appropriate here. Also, macros are the devil.